File: BiliRecommendUI/src/main/java/com/bilirecommendui/main/view/RecommendView.java
package com.bilirecommendui.main.view;


import com.bannerlayout.model.BannerModel;
import com.bilirecommendui.main.model.RecommendModel;

import java.util.List;

/**
 * by y on 2016/9/17.
 */
public interface RecommendView {

    void setBannerData(List<BannerModel> bannerData);

    void setRecommendData(RecommendModel recommendData);

    void removeAdapter();

    void netWorkError();

    void showProgress();

    void hideProgress();
}

File: BilibiliRecommendUi/app/src/main/java/com/bilibilirecommend/main/view/RecommendView.kt
package com.bilibilirecommend.main.view


import com.bannerlayout.model.BannerModel
import com.bilibilirecommend.main.model.RecommendModel

interface RecommendView {

    fun setBannerData(bannerData: List<BannerModel>)

    fun setRecommendData(recommendData: RecommendModel)

    fun removeAdapter()

    fun netWorkError()

    fun showProgress()

    fun hideProgress()
}
